### PR TITLE
Iss#116 pixel to global coords

### PR DESCRIPTION
**Pull request by Hsenrab**

### CODE REVIEW

File: roboplot/core/camera/pi_camera.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Camera:
 
     def take_photo_at(self, camera_centre):
         with picamera.PiCamera() as camera:
-            camera.resolution = (200, 200)
+            camera.resolution = CAMERA_RESOLUTION
```
Comment (JackBuck) [inline roboplot/core/camera/pi_camera.py]: I think you mean `config.CAMERA_RESOLUTION` here

File: roboplot/imgproc/path_following.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import operator
+import roboplot.config as config
+import roboplot.imgproc.image_analysis as image_analysis
+
+
+def convert_to_global_coords(points, scan_direction, camera_position):
+
+    # Scale factors
+    x_scaling = config.X_PIXELS_TO_POINTS_SCALE  # (based on a 200x200 pixel photo of a 40mm square)
+    y_scaling = config.Y_PIXELS_TO_POINTS_SCALE  # (based on a 200x200 pixel photo of a 40mm square)
```
Comment (JackBuck) [inline roboplot/imgproc/path_following.py]: Given that these are coming from the `config` file, I don't think you want the comments here do you?

File: roboplot/config.py [hunk 3]
```diff
@@ -19,3 +19,11 @@
 # Debugging image paths
 debug_image_file_path = os.path.normpath(os.path.join(resources_dir, 'Challenge_2_Test_Images', 'HackspacePath_Sample3.png'))
 debug_output_folder = os.path.normpath(os.path.join(resources_dir, 'DebugImages'))
+
+
+# Camera constants
+X_PIXELS_TO_POINTS_SCALE = 0.2
+Y_PIXELS_TO_POINTS_SCALE = 0.2
```
Comment (JackBuck) [inline roboplot/config.py]: Would these be better named `x_pixels_to_millimetres_scale` and `y_pixels_to_millimetres_scale`?

File: testing/test_convert_coords.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        self.assertEqual(output, expected_output_points)
+
+    def testConvertWithDefaultParam(self):
+        points = [(0, 100), (13, 120), (85, 122), (27, 128), (150, 130)]
+        centre = (50, 40)
+
+        scan_direction = image_analysis.Direction.SOUTH
+        output = path_following.convert_to_global_coords(points, scan_direction, centre)
+        expected_output_points = [(50, 40), (52.6, 44), (67, 44.4), (55.4, 45.6), (80, 46)]
+        self.assertEqual(output, expected_output_points)
```
Comment (JackBuck) [inline testing/test_convert_coords.py]: I'm no unit testing guru and maybe you did this on purpose, but this test is testing more than one thing (i.e. it has more than one reason to change):
  - It is the only test which tests non-trivial values for `config.X_PIXELS_TO_POINTS` and `config.Y_PIXELS_TO_POINTS` (though this is not explicit)
  - It is testing to make sure we don't change the values `config.X_PIXELS_TO_POINTS`, `config.Y_PIXELS_TO_POINTS` and `config.CAMERA_RESOLUTION`

Further, I'm not convinced of the value of testing to keep values in the `config` file constant? Surely if we change them then we will have done it on purpose?

Comment (JackBuck) [inline testing/test_convert_coords.py]: Also, when I run the tests, this one fails!!!

File: roboplot/config.py (hunk 3 above)
Comment (Hsenrab) [inline roboplot/config.py]: Caps is standard for constants - I believe but yet to name change.

File: testing/test_convert_coords.py (hunk 4 above)
Comment (Hsenrab) [inline testing/test_convert_coords.py]: Yes I found this myself it passes if you run it on its own. Need to re specify the constants which I thinks satisfies all your problems with it.

**Comment by Hsenrab** [comment]
@JackBuck Can you check y changes are ok please.

**Pull request merged**